### ISSUE TITLE
Feature Request: Support Tajweed Color Quran مصحف التجويد الملون

### ISSUE THREAD

**Issue by animehunter123** [quran/quran_android#1353]
Asa, 

Longtime follower of your app, great job over the years Jazak Allah Khair.

Can you add the Tajweed Color Quran to your app? The official PDF (high resolution) is available from the below page:

https://ar.islamway.net/book/15763/%D9%85%D8%B5%D8%AD%D9%81-%D8%A7%D9%84%D8%AA%D8%AC%D9%88%D9%8A%D8%AF-%D8%A7%D9%84%D9%85%D9%84%D9%88%D9%86

This is a excellent quran, and the 4th edition (which does not have the missing kasrah Page 457 / Chapter 38 / Ayah 71 in the word Lilmalaika).

This would be my dream wish if Quran Android can support this Mushaf, as it is Othman font. Can you add this feature, or maybe allow custom pdf import support?

Jazak Allah Khair, and Ramadan Mubarek.

![Mushaf Mulawin - Color Tajweed Quran](https://user-images.githubusercontent.com/42163211/81886677-43220900-95d8-11ea-8216-7c2ca2084f02.png)

**Comment by murtraja** [comment]
#439

**Comment by murtraja** [comment]
@animehunter123 It is one of my dreams too :heart_eyes:

**Comment by animehunter123** [comment]
I dont think it is too hard to do this. The Image files are already the same resolution if you use the KSU University version. **Have the developers seen this request?** I am still waiting for a response isA.

**Comment by murtraja** [comment]
@animehunter123 If you're feeling adventurous, you can give this a try. I have an old database for the ayah mapping [here](https://github.com/murtraja/quran-android-images-helper/blob/master/static/databases/ayahinfo_1053.db). The pages are 776x1053 pixels. You can get the pages from the same repo [here](https://github.com/murtraja/quran-android-images-helper/tree/master/static/images_1053).

You can test the database by trying different safah parameters [here](https://quran-android-images-helper.herokuapp.com/?safah=3&images=1053). (Don't worry too much about the line inconsistency and horizontal offsets, as they have been fixed already, I haven't updated the db yet.)

As far as integrating this with Quran Android is concerned, one will have to either 

add a few conditions to change the url (https://quran-android-images-helper.herokuapp.com/static/images_1053/1.jpg for example) of the images as these aren't available on their servers, plus the url of the database during the initial asset fetching routine

OR 

simply copy the assets over, bundling it with the app and providing proper local URIs 

OR 

use Android Studio file manager to overwrite the existing files with Tajweed files. Moreover, pages 1 and 2 have different resolution (827x1158 pixels) so that might have to be handled as well.

Since the database's schema is exactly identical to that of Quran Android's schema, there shouldn't be any problem as far as highlighting/ayah touch detection/ayah playback is concerned, but I can't be sure.

**Comment by animehunter123** [comment]
Whoa.... Thank you for typing this for me. I will try to do this (I am not really a programmer but just a windows powershell guy, but I certainly will look over this. Looks like it is all on github which is kind of cool!)

Whew... still seems scary but I will try...

**Comment by animehunter123** [comment]
By the way, the https://github.com/murtraja/quran-android-images-helper/tree/master/static/images_1053 is using the "Spaced Color Tajweed Mushaf", but what we really want is the Mushaf Mulawen without spaces which is from the URL above (https://ar.islamway.net/book/15763/%D9%85%D8%B5%D8%AD%D9%81-%D8%A7%D9%84%D8%AA%D8%AC%D9%88%D9%8A%D8%AF-%D8%A7%D9%84%D9%85%D9%84%D9%88%D9%86)

Man... I feel intimidated trying to do this but iSA I can figure it out...

**Comment by animehunter123** [comment]
Ok, making some small progress, I will continue to add my progress here. I was able to install android studio, and git pull the project into it. I got it to successfully compile and run in the android sdk pixel vm, now I want to add my images that I have resized to the package. They are the same as the default one in your database (these images are from the PDF above, but cropped by Ayat KSU so that they do not include the side area, so they are almost exactly the same as the Ayat). 

How do I upload them to the fork so that insha Allah they can be included in the baseline eventually?

**Comment by murtraja** [comment]
> How do I upload them to the fork so that insha Allah they can be included in the baseline eventually?

The easiest thing for us would be to simply upload the images to official quran android servers, like http://android.quran.com/data/width_1260/page120.png , but unfortunately, we don't have access to those servers. The good news is that we don't need access in order to test it. Uploading tajweed images to the official servers will be done by maintainers once we submit a PR, it should be the last step.

And hence, I have already mentioned 3 methods in which we can display the tajweed pages in the app (for now)
1. Upload images to your server, point to it in the app (difficulty: advanced)
2. Bundle the images in the app itself (no downloading from external servers, difficulty: moderate)
3. After original images get downloaded, overwrite those with tajweed images along with the database(This is by far the easiest method)

Once, the audio, the highlighting, etc. work correctly, we can submit a PR and work on the suggestions posted by the maintainers.

**Comment by animehunter123** [comment]
Jazak Allah Khair. Thank you for typing this out for me, sorry for the basic questions. 

I am playing around with option #3, and learned how to get gradle to make the apk files to test on my phone. It was not too hard at all like you said (obviously highlighting and other things didnt look right, but it "is working" yay!

Understand and will continue to play with it. Kind regards, and thank you so much.

**Comment by mouhanad1053** [comment]
بسم الله الرحمن الرحيم In the name of Allah, Most Gracious, Most Merciful

ASA,

Following up to the comment above, I now have a Beta Flutter web app at https://www.indie-publishing.space/self_publish_app/#/. Some features are still under development.

Any comment is greatly appreciated.
